docs/determinism.md

```diff
@@ -0,0 +1,84 @@
+# Determinism 🤝 Rollups
+
+[Determinism](https://en.wikipedia.org/wiki/Deterministic_system#In_computer_science) is crucial for rollups because of [fault proof](https://community.optimism.io/docs/protocol/2-rollup-protocol/#fault-proofs). Validators will execute program([Turing machine](https://en.wikipedia.org/wiki/Turing_machine) tape) to validate the proposed state of rollup, syncing shared [repliced state machine](https://en.wikipedia.org/wiki/State_machine_replication). The state machine consists of a transition function. The transition function must be deterministic to decide which party is correct when dispute occur. For example, let's say that Alice is the state proposer and Bob is a validator. Bob will advance its own machine state by validating the program, and checks the sanity of proposed state at L1 and disagrees with it, and tells L1 that Alice has lied, triggering fault proof. If the program is nondeterministic, L1 will never find out who's claim(state) is right, leading L1 security inheritance loss.
```

Comment:
First Sentence:
- The first sentence is not clear. No clear link between the two referenced sources.
- The determinism in the reference doesn't explain the determinism spoken in the Ethereum world.
- The reference for the fault proof doesn't clearly explain why determinism is important in rollups.
Second Sentence
- Why do validators run Turing machines? Aren't all general-purpose computers Turing machines? What's the importance of mentioning Turing machines?